### ISSUE TITLE
findAllVNodes doesn't look through children

### ISSUE THREAD

**Issue by andreasvirkus** [eddyerburgh/avoriaz#3]
Hello again! It seems for my setup that `.find()` doesn't find any child nodes and therefore only looks at the parent node, making `.find()` the equivalent of `.hasClass()` (when used with a class selector). When used with an ID, it gives the error 

     TypeError: node.elm.getAttribute is not a function

Also, in vNode.js, why is there a check for `nodeName === '#text'` (findById, line 32)?

The template of my component:
```
<template>
  <transition name="modal">
    <div class="modal-mask" @click="hideModal" transition="modal">
      <div class="modal-dialog" @click.stop>
        <div class="modal-header">
            <a class="btn close" @click="hideModal">
                <i class="icon-cross color-darkest-gray large"></i>
            </a>
        </div>

        <div class="modal-body">
            <slot name="body">
                <p>Default modal content goes here.</p>
            </slot>
        </div>
      </div>
    </div>
  </transition>
</template>

<script>
import { mapActions } from 'vuex';

export default {
    name: 'modal',
    methods: {
      ...mapActions('Modals', [
        'hideModal'
      ])
    }
}
</script>
```

And when running 
```
import { mount } from 'avoriaz';
import { expect } from 'chai';
import Modal from '../src/components/Modal.vue';

describe('Modal.vue', () => { 
  it('renders a div element with class modal-mask', () => {
    const modal = mount(Modal);

    expect(modal.is('div')).to.equal(true);
    expect(modal.hasClass('modal-mask')).to.equal(true); // Here we can see the component considers this to be its root node
  });

  it('renders a div element with class modal-mask', () => {
    const modal = mount(Modal);
    const mask = modal.find('.modal-mask')[0]; // Finds .modal-mask

    expect(mask.is('div')).to.equal(true);
  });

  it('renders a div element with class modal-dialog', () => {
    const modal = mount(Modal);
    const dialog = modal.find('.modal-dialog')[0]; 
    // .find() returns empty array, dialog equals undefined at this point
    // Same happens for any child node

    expect(dialog.is('div')).to.equal(true);
  });
});
```

Should I try and replicate this in a fiddle?

**Comment by eddyerburgh** [comment]
Thanks @andreasvirkus .

The reason there is an if statement for #text is to stop errors thrown by text nodes when checking if vNodes have an id. In Vue text nodes are rendered as vNodes with nodename #text. 

No need to replicate, I have enough info :+1: 

I'll look into this either this afternoon or tomorrow and get a much more robust vnode traversal set up.

**Comment by andreasvirkus** [comment]
when using `util.inspect`, I could find some references of the child elements under
_vnode (maybe it's of some extra help):

```
_vnode:
      { tag: 'div',
        data:
         { staticClass: 'modal-mask',
           attrs: { transition: 'modal' },
           on:
            { click: { fn: [Object], invoker: [Function] },
              keyup: { fn: [Function: keyup], invoker: [Function] } },
           transition: { name: 'modal' } },
        children:
         [ { tag: 'div',
             data: { staticClass: 'modal-dialog', on: [Object] },
             children: [ [Object], [Object], [Object] ],
             text: undefined,
             elm: { _prevClass: 'modal-dialog' },
```

**Comment by eddyerburgh** [comment]
Hey, so I've just got round to having a look. The vNode tree is different for transitions, I haven't tested for them. Tomorrow I'm going to change how the vNode search works to account for transitions and scopes.

In the meantime - here's one way to test the div is being rendered:

```
  it('renders a div element with class modal-dialog', () => {
    const modal = mount(Modal);
    console.log(modal);
    const modalDialog = modal.element.querySelector('.modal-dialog');
    expect(modalDialog.tagName).to.equal('DIV');
  });
```

**Comment by eddyerburgh** [comment]
Should be fixed with this commit 66ca264d62765affb05b9bf3487bc243d0fcdef4 .

I've published a new version - 1.2.3 - let me know if it fixes your issue :smile:

**Comment by andreasvirkus** [comment]
Cheers, the fix works fine!
